### PR TITLE
Cleanup empty files

### PR DESCRIPTION
**Pull request by Echelon9**
FS2Open contains a number of redundant empty C++ and header files.

This PR removes the below empty files, references to them and the associated changes required to project files (see below):
* ai/ailocal.h
* missionui/missionrecommend.cpp
* missionui/missionrecommend.h
* missionui/missionstats.cpp
* missionui/missionstats.h

Makefile and Xcode4+ project files have been updated. Could a Windows developer push a commit to my branch with updated MSVC project files?

### CODE REVIEW

**Comment by The-E** [comment]
There's a pull request on your branch for that

**Comment by Echelon9** [comment]
Thanks @The-E

File: projects/MSVC_2012/code.vcxproj [hunk 1]
```diff
@@ -160,6 +160,7 @@
       <Optimization>Disabled</Optimization>
       <AdditionalIncludeDirectories>../../code;../../libjpeg;../../libpng;../../lua;../../oggvorbis/include;../../openal/include;../../zlib;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
       <PreprocessorDefinitions>_WINDOWS;WIN32;_DEBUG;PDB_DEBUGGING;_CRT_SECURE_NO_DEPRECATE;_SECURE_SCL=0;_HAS_ITERATOR_DEBUGGING=0;%(PreprocessorDefinitions)</PreprocessorDefinitions>
+      <MinimalRebuild>true</MinimalRebuild>
```
Comment (asarium) [inline projects/MSVC_2012/code.vcxproj]: Could you remove this again? This will break multi-processor compilation and doesn't benefit us much.

**Comment by chief1983** [comment]
The MSVC changes look like they affect way more than necessary for this, I think they need to be backed out and re-done from scratch.

**Pull request merged**